decred/tests/unit/dcr/test_txscript.py

```diff
@@ -2352,30 +2353,86 @@ def test_scriptNumBytes(self):
+            """
+            self.name = name
+            self.pkScript = pkScript
+            self.wantException = wantException
+            self.want = want
+
+    # fmt: off
+    tests = [
+        test(
+            name="P2PKH",
+            pkScript=ByteArray([opcode.OP_DUP, opcode.OP_HASH160,
+                                opcode.OP_DATA_20, *([0x00] * 20),
+                                opcode.OP_EQUALVERIFY, opcode.OP_CHECKSIG]),
+            wantException=None,
+            want=txscript.RedeemP2PKHSigScriptSize,
+        ),
+        test(
+            name="P2PK",
+            pkScript=ByteArray([opcode.OP_DATA_33, 0x02, *([0x00] * 32),
+                                opcode.OP_CHECKSIG]),
+            wantException=None,
+            want=txscript.RedeemP2PKSigScriptSize,
+        ),
+        test(
+            name="revocation",
+            pkScript=ByteArray([opcode.OP_SSRTX, opcode.OP_DUP, opcode.OP_HASH160,
+                                opcode.OP_DATA_20, *([0x00] * 20),
+                                opcode.OP_EQUALVERIFY, opcode.OP_CHECKSIG]),
+            wantException=None,
+            want=txscript.RedeemP2PKHSigScriptSize,
+        ),
+        test(
+            name="stake change",
+            pkScript=ByteArray([opcode.OP_SSTXCHANGE, opcode.OP_DUP, opcode.OP_HASH160,
+                                opcode.OP_DATA_20, *([0x00] * 20),
+                                opcode.OP_EQUALVERIFY, opcode.OP_CHECKSIG]),
+            wantException=None,
+            want=txscript.RedeemP2PKHSigScriptSize,
+        ),
+        test(
+            name="stake gen",
+            pkScript=ByteArray([opcode.OP_SSGEN, opcode.OP_DUP, opcode.OP_HASH160,
+                                opcode.OP_DATA_20, *([0x00] * 20),
+                                opcode.OP_EQUALVERIFY, opcode.OP_CHECKSIG]),
+            wantException=None,
+            want=txscript.RedeemP2PKHSigScriptSize,
+        ),
+        test(
+            name="unsupported stake submission",
+            pkScript=ByteArray([opcode.OP_SSTX, opcode.OP_DUP, opcode.OP_HASH160,
+                                opcode.OP_DATA_20, *([0x00] * 20),
+                                opcode.OP_EQUALVERIFY, opcode.OP_CHECKSIG]),
+            wantException=NotImplementedError,
+            want=None,
+        ),
+        test(
+            name="unsupported nested script",
+            pkScript=ByteArray([opcode.OP_SSRTX, opcode.OP_HASH160,
+                                opcode.OP_DATA_20, *([0x00] * 20), opcode.OP_EQUAL]),
+            wantException=DecredError,
+            want=None,
+        ),
+    ]
+    # fmt: on
+    for test in tests:
+        if test.wantException:
+            with pytest.raises(test.wantException):
+                txscript.spendScriptSize(test.pkScript)
+            continue
+        res = txscript.spendScriptSize(test.pkScript)
+        assert (
+            res == test.want
+        ), f"wanted {test.want} for test {test.name} but got {res}"
```

Comment:
Same comment as in #89.

Using a dict instead of a class for test data would probably do away with the need for custom failure messages, because pytest shows the data in the local context. A dict would be shown with all items data visible, while an instance would not show its attributes.

Reply:
ok, you sold me.

Reply:
ah, left in the custom message.